pwiz_tools/Skyline/Model/Proteome/ProteinGroupMetadata.cs

```diff
@@ -0,0 +1,219 @@
+﻿/*
+ * Original author: Matt Chambers <matt.chambers42 .@. gmail.com>
+ *
+ * Copyright 2022
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using pwiz.Common.Collections;
+using pwiz.Common.SystemUtil;
+using pwiz.ProteomeDatabase.API;
+using pwiz.Skyline.Util;
+
+namespace pwiz.Skyline.Model.Proteome
+{
+    public class ProteinGroupMetadata : ProteinMetadata
+    {
+        const string GROUP_SEPARATOR = @"/";
+        public new static readonly ProteinGroupMetadata EMPTY = new ProteinGroupMetadata();
+
+        private ProteinGroupMetadata() : base(null, null)
+        {
+        }
+
+        private ProteinGroupMetadata(ProteinGroupMetadata other, WebSearchInfo webSearchInfo = null) : base(other.Name, other.Description)
+        {
+            webSearchInfo ??= other.ProteinMetadataList.First().WebSearchInfo;
+            ProteinMetadataList = ImmutableList<ProteinMetadata>.ValueOf(other.ProteinMetadataList);
+            //ProteinMetadataList[0] = ProteinMetadataList[0].ChangeWebSearchInfo(webSearchInfo);
+            /*    Name = ProteinMetadataList.All(p => p.Name == null)
+                ? null
+                : string.Join(GROUP_SEPARATOR, ProteinMetadataList.OrderBy(p => p.).Select(p => p.Name));*/
+        }
+
+        public ProteinGroupMetadata(string name, string description, IList<ProteinMetadata> proteinMetadata) : base(name, description)
```

Comment:
I still don't understand why this class has a ctor that takes name and description when it doesn't take all the others. I get that "all the others" are aggregates from the proteins in the group, but so are Name and Description if I'm reading the code correctly. Seems like the ctor name and description are unrecoverable?

Reply:
Yeah, that is pretty pointless. I'll make it just take the IList and pass null to the base class.